_tutorials/2016-07-18-python-automation-netmiko-and-napalm.md

```diff
@@ -0,0 +1,226 @@
+---
+published: false
+date: '2016-07-18 23:11 +0300'
+title: 'Python automation: netmiko and napalm'
+author: Mike Korshunov
+tags:
+  - vagrant
+  - iosxr
+  - Python
+excerpt: Brief introduction to automation with Python
+---
+{% include toc icon="table" title="IOS-XR: Ansible and Vagrant" %}
+
+
+
+## Introduction
+
+At first, let's take a look at used tools:
+
+[Netmiko](https://github.com/ktbyers/netmiko) - multi-vendor ssh tool for device configuration 
+
+[NAPALM](https://github.com/napalm-automation/napalm)- tool, which provides standardized API for different platforms. 
+
+We will use old setup, which consist of devbox (Ubuntu instance) and rtr (IOS-XRv)  [link]({{ site.url }}/tutorials/IOSXR-Ansible).
+{: .notice--info}  
+
+We are interesting in IOS-XRv in particular. 
+
+### Module installation
+
+Let's install python modules on devbox:
+
+```shell
+sudo pip install netmiko
+
+sudo pip install napalm
+```
+
+As first step we can verify, that setup is successful. 
```

Comment:
To begin with   OR  to start off,  we should verify that the setup was successful.